src/code/z_camera.c

```diff
@@ -641,62 +641,56 @@ s16 func_80044ADC(Camera* camera, s16 yaw, s16 arg2) {
     return temp_s0 + temp_s1;
 }
 
-Vec3f* Camera_CalcUpFromPitchYawRoll(Vec3f* dest, s16 pitch, s16 yaw, s16 roll) {
-    f32 sinPitch;
-    f32 cosPitch;
-    f32 sinYaw;
-    f32 cosYaw;
-    f32 sinNegRoll;
-    f32 cosNegRoll;
-    Vec3f spA4;
+/**
+ * Calculates a new Up vector from the pitch, yaw, roll
+ */
+Vec3f* Camera_CalcUpFromPitchYawRoll(Vec3f* viewUp, s16 pitch, s16 yaw, s16 roll) {
```

Comment:
Afaict but this needs testing, pitch and roll are of the opposite sign of what you'd expect if they were named like rotX and rotZ

So I expect ascending pitch to aim upwards, ascending roll to make up point left (making it look like the world in front moves clockwise... I think)
(and as "normal" ascending yaw to go left)

So I tested and yeah that's right